File: .scripts/generate_sitemap.py
import requests
import xml.etree.ElementTree as ET
from datetime import datetime

# Configuration
REPO_OWNER = "YayaBed"
REPO_NAME = "MagneticHub.fr"
BRANCH = "main"
ARTICLES_DIR = "articles"
BASE_URL = "https://yayabed.github.io/MagneticHub.fr"
SITEMAP_FILE = "sitemap.xml"

def get_github_file_list():
    api_url = f"https://api.github.com/repos/{REPO_OWNER}/{REPO_NAME}/contents/{ARTICLES_DIR}?ref={BRANCH}"
    response = requests.get(api_url)
    if response.status_code != 200:
        raise Exception(f"Erreur de récupération des fichiers depuis GitHub API: {response.status_code}")
    return response.json()

def generate_sitemap(entries):
    urlset = ET.Element("urlset", xmlns="http://www.sitemaps.org/schemas/sitemap/0.9")
    for entry in entries:
        if entry["name"].endswith(".md"):
            slug = entry["name"].replace(".md", "")
            url = f"{BASE_URL}/{slug}.html"  # suppose que GitHub Pages rend les .md en .html
            url_el = ET.SubElement(urlset, "url")
            ET.SubElement(url_el, "loc").text = url
            ET.SubElement(url_el, "lastmod").text = datetime.utcnow().date().isoformat()
            ET.SubElement(url_el, "changefreq").text = "weekly"
            ET.SubElement(url_el, "priority").text = "0.8"
    return ET.tostring(urlset, encoding="utf-8", xml_declaration=True).decode("utf-8")

def save_sitemap(xml_content):
    with open(SITEMAP_FILE, "w", encoding="utf-8") as f:
        f.write(xml_content)
    print(f"✅ sitemap.xml mis à jour avec {xml_content.count('<url>')} URLs.")

if __name__ == "__main__":
    files = get_github_file_list()
    sitemap_xml = generate_sitemap(files)
    save_sitemap(sitemap_xml)

File: .github/workflows/generate-sitemap.js
const fs = require('fs');
const path = require('path');

// Répertoire des pages HTML (dossier racine du projet GitHub Pages)
const baseDir = path.join(__dirname, 'dist'); // remplace par '.' si à la racine
const baseUrl = 'https://yayabed.github.io/MagneticHub.fr';

const getAllHtmlFiles = (dir, fileList = []) => {
  const files = fs.readdirSync(dir);
  files.forEach(file => {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      getAllHtmlFiles(fullPath, fileList);
    } else if (file.endsWith('.html')) {
      fileList.push(fullPath);
    }
  });
  return fileList;
};

const formatDate = () => {
  return new Date().toISOString();
};

const generateSitemap = () => {
  const pages = getAllHtmlFiles(baseDir);
  const urls = pages.map(pagePath => {
    const relativePath = path.relative(baseDir, pagePath).replace(/\\/g, '/');
    const loc = `${baseUrl}/${relativePath}`.replace(/index\.html$/, '');
    return `
  <url>
    <loc>${loc}</loc>
    <lastmod>${formatDate()}</lastmod>
    <priority>0.80</priority>
  </url>`;
  });

  const sitemap = `<?xml version="1.0" encoding="UTF-8"?>  
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
${urls.join('\n')}
</urlset>`;

  fs.writeFileSync(path.join(baseDir, 'sitemap.xml'), sitemap, 'utf8');
  console.log('✅ Sitemap généré avec succès.');
};

generateSitemap();
